src/fidesops/models/privacy_request.py

```diff
@@ -396,44 +440,47 @@ def error_processing(self, db: Session) -> None:
 PAUSED_SEPARATOR = "__fidesops_paused_sep__"
 
 
-def cache_step_and_collection(
-    step: Optional[PausedStep], collection: Optional[CollectionAddress], cache_key: str
+def cache_restart_details(
+    cache_key: str,
+    step: Optional[PausedStep] = None,
+    collection: Optional[CollectionAddress] = None,
+    action_needed: Optional[List[ManualAction]] = None,
 ) -> None:
-    """Generic method to cache which collection and which step (access/erasure) we're stopped on
+    """Generic method to cache which collection and step(access/erasure) halted PrivacyRequest execution,
+    as well as data (action_needed) to resume the privacy request if applicable.
 
-    For example, we might pause a privacy request at the erasure step of the postgres_example:address collection or
-    the request might have failed at the access step of the mongo_example:payment_card collection.
+    For example, we might pause a privacy request at the access step of the postgres_example:address collection.  The
+    user might need to retrieve an "email" field and an "address" field where the customer_id is 22 to resume the request.
 
     The "step" is needed because we may build and execute multiple graphs as part of running a privacy request.
     An erasure request builds two graphs, one to access the data, and the second to mask it.
     We need to know if we should resume execution from the access or the erasure portion of the request.
     """
     cache: FidesopsRedis = get_cache()
+    cache_stopped: Optional[StoppedCollection] = None
+    if collection and step:
+        cache_stopped = StoppedCollection(
+            step=step, collection=collection, action_needed=action_needed
+        )
 
-    # Store both the step and collection under one value
     cache.set_encoded_object(
         cache_key,
-        f"{step.value}{PAUSED_SEPARATOR}{collection.value}"
-        if step and collection
-        else None,
+        cache_stopped.dict() if cache_stopped else None,
     )
```

Comment:
This caches paused/failed collection details in a different format than was released previously, storing multiple details in a dictionary instead of a separated string.  I doubt anyone is using this yet, so I think it's okay to swap this format.